air/src/stack/field_ops/mod.rs

```diff
@@ -59,25 +55,76 @@ pub fn enforce_constraints<E: FieldElement>(
 ) -> usize {
     let mut index = 0;
 
+    // Enforce constaints of the EQZ operation.
+    index += enforce_eqz_constraints(frame, result, op_flag.eqz());
+
     // Enforce constaints of the INCR operation.
-    index += enforce_incr_constraints(frame, result, op_flag.incr());
+    index += enforce_incr_constraints(frame, &mut result[index..], op_flag.incr());
+
     // Enforce constaints of the INV operation.
     index += enforce_inv_constraints(frame, &mut result[index..], op_flag.inv());
+
     // Enforce constaints of the NEG operation.
     index += enforce_neg_constraints(frame, &mut result[index..], op_flag.neg());
+
     // Enforce constaints of the NOT operation.
     index += enforce_not_constraints(frame, &mut result[index..], op_flag.not());
 
+    // Enforce constaints of the ADD operation.
+    index += enforce_add_constraints(frame, &mut result[index..], op_flag.add());
+
+    // Enforce constaints of the MUL operation.
+    index += enforce_mul_constraints(frame, &mut result[index..], op_flag.mul());
+
+    // Enforce constaints of the AND operation.
+    index += enforce_and_constraints(frame, &mut result[index..], op_flag.and());
+
+    // Enforce constaints of the OR operation.
+    index += enforce_or_constraints(frame, &mut result[index..], op_flag.or());
+
+    // Enforce constaints of the EQ operation.
+    index += enforce_eq_constraints(frame, &mut result[index..], op_flag.eq());
+
     index
 }
 
 // TRANSITION CONSTRAINT HELPERS
 // ================================================================================================
 
+/// Enforces constraints of the EQZ operation. The EQZ operation checks if the top element in the
+/// current frame is 0 or not. Therefore, the following constraints are enforced:
+/// - s0 * s0` = 0.
+/// - s0` - (1 - h0 * s0) = 0.
+pub fn enforce_eqz_constraints<E: FieldElement>(
```

Comment:
Is there a reason to put this first but `eq` constraints last? In general, I think we should follow the order we have in the docs (i.e., [here](https://maticnetwork.github.io/miden/design/stack/field_ops.html)). For example, `add`, `neg`, `mul` etc. This would make things easier to review/audit.

Reply:
I was following the order in from [here](https://maticnetwork.github.io/miden/design/stack/op_constraints.html#no-stack-shift-operations). I've updated the order now.